packages/flutter_tools/lib/src/base/version.dart

```diff
@@ -36,8 +34,8 @@ class Version implements Comparable<Version> {
   }
 
   /// Creates a new [Version] by parsing [text].
-  factory Version.parse(String text) {
-    final Match match = versionPattern.firstMatch(text ?? '');
+  static Version? parse(String? text) {
```

Comment:
Null safety guide says don't return null from factories.  Convert to a static method instead per https://dart.dev/null-safety/faq#how-should-i-migrate-a-factory-that-can-return-null